tutorial/lesson_22_serialization.cpp

```diff
@@ -0,0 +1,169 @@
+// Halide tutorial lesson 22: Serialization
+
+// This lesson describes how to serialize pipelines into a binary format 
+// which can be saved on disk, and later deserialized and loaded for 
+// evaluation.
```

Comment:
We should add a note to the effect that "Serialization is *experimental* in Halide 17 and is subject to change; we recommend that you avoid relying on it for production work at this time". (On that note: I wonder if we should consider moving the API into something like `Halide::Experimental` namespace?)

Reply:
Okay!  I'll add update the comments.  Having an experimental namespace seems like a worthwhile thing ... maybe worth a separate PR.